### PR TITLE
Reference a specific devtools binary version

### PR DESCRIPTION
**Pull request by jlindbergh**
This simplifies the version checking and also enables manual preinstallation of the expected devtools binary.

If download fails, we use the status-view to provide the user with an actionable message explaining how to download and install the binary manually (putting it in the `extensionPath`)

### CODE REVIEW

File: src/download.ts [hunk 1]
```diff
@@ -15,140 +12,99 @@ import * as path from 'path';
...
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const EXPECTED_DEVTOOLS_VERSION = 'c0b212c6be46a95802666aaf0a18678b055feb1e';
+
 const artifacts: { [platform: string]: { [arch: string]: string } } = {
   darwin: {
-    x64: 'codescene-cli-ide-macos-amd64-v4.zip',
-    arm64: 'codescene-cli-ide-macos-aarch64-v4.zip',
+    x64: `codescene-cli-ide-macos-amd64-${EXPECTED_DEVTOOLS_VERSION}.zip`,
+    arm64: `codescene-cli-ide-macos-aarch64-${EXPECTED_DEVTOOLS_VERSION}.zip`,
```
Comment (TobiasMakitalo) [inline src/download.ts:27]: Maybe should be called required dectools version?

File: src/download.ts [hunk 2]
```diff
@@ -159,54 +115,44 @@ function download(url: URL, filePath: string) {
...
-async function ensureBinaryRuns(cliPath: string): Promise<string> {
-  const signResult = await new SimpleExecutor().execute({ command: cliPath, args: ['version'], ignoreError: true }, {});
-
-  if (signResult.exitCode === 0) {
-    outputChannel.appendLine('The latest CodeScene CLI is in place, we are ready to go!');
-    return cliPath;
-  } else {
-    throw new Error(`Error invoking the downloaded CLI binary: ${signResult.stderr}`);
-  }
+async function ensureBinaryRuns(cliPath: string) {
```
Comment (TobiasMakitalo) [inline src/download.ts:118]: Maybe not needed when we also run isExpectedCliVersion when validating the installed binary

**Review by TobiasMakitalo — COMMENTED** [review]
We should try to put together a manual end-to-end test for this

File: src/download.ts (hunk 1 above)
Comment (jlindbergh) [reply src/download.ts:27]: Yeah thanks, I got stuck in naming limbo for a while there...

File: src/download.ts (hunk 2 above)
Comment (jlindbergh) [reply src/download.ts:118]: Yes, but we need to ensure it runs after being downloaded as well. `ensureBinaryRuns` was originally intended for catching the rare "missing instruction set" error so it was simplest to just reuse that.
I remember thinking about just running `isExpectedCliVersion` again, maybe I'll do that and just throw a general invocation error if it returns false.

**Pull request merged**